readthedocs/projects/views/private.py

```diff
@@ -674,3 +674,24 @@ class DomainUpdate(DomainMixin, UpdateView):
 
 class DomainDelete(DomainMixin, DeleteView):
     pass
+
+
+@login_required
+def project_resync_webhook(request, project_slug):
+    """
+    Resync a project webhook.
+
+    The signal will add a success/failure message on the request.
+    """
+    project = get_object_or_404(Project.objects.for_admin_user(request.user),
+                                slug=project_slug)
+    if request.method == 'POST':
+        project_import.send(sender=project, request=request)
+        return HttpResponseRedirect(reverse('projects_detail',
+                                            args=[project.slug]))
+
+    return render_to_response(
+        'projects/project_resync_webhook.html',
+        {'project': project},
+        context_instance=RequestContext(request)
+    )
```

Comment:
Agreed. It should probably just be AJAX, but I don't think we have a pattern for it. Opened #2034